### PR TITLE
Add label_shapes parameter to KerasEstimator and TorchEstimator

### PR DESCRIPTION
**Pull request by abditag2**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: horovod/spark/common/util.py [hunk 1]
```diff
@@ -171,24 +171,38 @@ def check_shape_compatibility(metadata, feature_columns, label_columns,
...
+            target_size = abs(np.prod(target_shape))
+            if _col_size != target_size:
                 raise ValueError('Label column \'{col}\' with size {label} must equal that of the '
-                                 'model output at index {idx} with size {output}'
-                                 .format(col=col, label=col_size, idx=idx, output=output_size))
+                                 'model output and label shape at index {idx} with size {output}'
+                                 .format(col=_col, label=_col_size, idx=_idx, output=target_size))
+
+    if output_shapes is not None:
+        _check_label_cols_size(output_shapes)
```
Comment (tgaddair) [inline horovod/spark/common/util.py:202]: Seems this should only be done if `label_shapes is None`, right?  Otherwise, we will be forcing the model output size to equal the label size, which may not be correct (for these custom loss functions).

File: horovod/spark/torch/remote.py [hunk 2]
```diff
@@ -266,9 +267,16 @@ def transform_outputs(outputs, labels):
 
                         # reshape labels to match the output shape of the model
                         if hasattr(outputs[0], 'shape'):
-                            labels = [label.reshape(output.shape)
-                                      if output.shape.numel() == label.shape.numel() else label
-                                      for label, output in zip(labels, outputs)]
+                            # If label_shapes parameter is not provided, reshape the label columns
```
Comment (tgaddair) [inline horovod/spark/torch/remote.py:270]: Can move this comment into the `else` branch so it's closer to the code it's describing.
